### PR TITLE
add the ability to profile Posix applications with gprof

### PR DESCRIPTION
**Pull request by alfred2g**
add the ability to profile Posix applications with gprof

Description
-----------
A user who ported his application to FreeRTOS Posix can now profile his application with gprof
they can also run their application with sanitizers enabled

Test Steps
-----------
```
make PROFILE=1
```
after running the application a gmon.out file will be generated.
the file can be converted with
```
make profile
```
in order to generate prof_flat.txt and prof_call_graph.txt

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by paulbartell** [comment]
@alfred2g : I pushed a commit to remove trailing whitespace from the makefile. 
This PR looks good overall. Could you add a CI job to the github workflow so that this doesn't bit rot?

**Comment by alfred2g** [comment]
@paulbartell : Thanks, We already have a job that runs posix "Core-Checker / FreeRTOS Posix Build Check (pull_request)"

File: FreeRTOS/Demo/Posix_GCC/Readme.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+```
+$ make PROFILE=1
+```
+Run your application
+```
+$ ./build/possix_demo
+```
+Since FreeRTOS and its application never come to an end and typically run
+forever. The user has to kill the application with **Ctrl_C**  when they feel
+satisfied that the application achieved its indented task. Killing the
```
Comment (paulbartell) [inline FreeRTOS/Demo/Posix_GCC/Readme.md:29]: s/indented/intended/

File: FreeRTOS/Demo/Posix_GCC/Readme.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+application will force the profiling file *gmon.out* to be generated
+automatically.
+In order to make sense of this file, the user has to convert the file with:
+```
+$ make profile
+```
+two (2) files *prof_call_graph.txt* and *prof_flat.txt* will be generated in the
+build directory.
+In order to understand the outputs generated, the best way is to read the
```
Comment (paulbartell) [inline FreeRTOS/Demo/Posix_GCC/Readme.md:38]: This sentence could be rewritten for clarity.

File: FreeRTOS/Demo/Posix_GCC/main.c [hunk 3]
```diff
@@ -231,13 +250,13 @@ void vApplicationTickHook( void )
...
-    int ret;
+    int xReturn;
     struct timeval tv = { 0L, 0L };
     fd_set fds;
     FD_ZERO(&fds);
     FD_SET(0, &fds);
-    ret = select(1, &fds, NULL, NULL, &tv);
-    if ( ret > 0 )
+    xReturn = select(1, &fds, NULL, NULL, &tv);
+    if ( xReturn > 0 )
```
Comment (paulbartell) [inline FreeRTOS/Demo/Posix_GCC/main.c:259]: What's going on with the indentation here? Have you run this through uncrustify?

**Comment by paulbartell** [comment]
> @paulbartell : Thanks, We already have a job that runs posix "Core-Checker / FreeRTOS Posix Build Check (pull_request)"

Yes. The CI job builds the demo, but does not run it.

Could you construct a job that runs the demo for a short amount of time to take advantage of the tests/sanitizers and/or verifies that the grpof files get output correctly?

Perhaps it could be a backlog item.

File: FreeRTOS/Demo/Posix_GCC/main.c (hunk 3 above)
Comment (alfred2g) [reply FreeRTOS/Demo/Posix_GCC/main.c:259]: nope i forgot :) , i will do it

**Comment by alfred2g** [comment]
> > @paulbartell : Thanks, We already have a job that runs posix "Core-Checker / FreeRTOS Posix Build Check (pull_request)"
> 
> Yes. The CI job builds the demo, but does not run it.
> 
> Could you construct a job that runs the demo for a short amount of time to take advantage of the tests/sanitizers and/or verifies that the grpof files get output correctly?
> 
> Perhaps it could be a backlog item.

yeah, backlog item would be better, we can discuss that.

**Review by paulbartell — DISMISSED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**